### PR TITLE
Prevent render InputHelperText if helperText props is false in Input components

### PR DESCRIPTION
**Pull request by adguernier**
related to #8830 and #8829

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx [hunk 1]
```diff
@@ -72,11 +73,13 @@ export const NullableBooleanInput = (props: NullableBooleanInputProps) => {
             }
             error={(isTouched || isSubmitted) && invalid}
             helperText={
-                <InputHelperText
-                    touched={isTouched || isSubmitted}
-                    error={error?.message}
-                    helperText={helperText}
-                />
+                helperText ? (
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx:76]:
```suggestion
                renderHelperText ? (
```

**Review by djhi — APPROVED** [review]

File: packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx (hunk 1 above)
Comment (adguernier) [reply packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx]: Wrong copy / paste :see_no_evil:

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 2]
```diff
@@ -537,6 +537,9 @@ If you provided a React element for the optionText prop, you must also provide t
         return String(getChoiceValue(option)) === String(getChoiceValue(value));
     };
 
+    const renderHelperText =
+        helperText !== false || ((isTouched || isSubmitted) && invalid);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:541]: You're missing `fetchError` in this condition

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx [hunk 3]
```diff
@@ -199,6 +199,9 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
         );
     }
 
+    const renderHelperText =
+        helperText !== false || ((isTouched || isSubmitted) && invalid);
+
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:204]: You're missing `fetchError` in the condition

File: packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx [hunk 4]
```diff
@@ -158,6 +158,10 @@ export const RadioButtonGroupInput = (props: RadioButtonGroupInputProps) => {
             </Labeled>
         );
     }
+
+    const renderHelperText =
+        helperText !== false || ((isTouched || isSubmitted) && invalid);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/RadioButtonGroupInput.tsx:163]: You're missing `fetchError` in the condition

File: packages/ra-ui-materialui/src/input/SelectArrayInput.tsx [hunk 5]
```diff
@@ -274,6 +274,8 @@ export const SelectArrayInput = (props: SelectArrayInputProps) => {
                   ),
               }
             : {};
+    const renderHelperText =
+        helperText !== false || ((isTouched || isSubmitted) && invalid);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.tsx:278]: You're missing `fetchError` in the condition

File: packages/ra-ui-materialui/src/input/SelectInput.tsx [hunk 6]
```diff
@@ -296,6 +296,9 @@ export const SelectInput = (props: SelectInputProps) => {
         );
     }
 
+    const renderHelperText =
+        helperText !== false || ((isTouched || isSubmitted) && invalid);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/SelectInput.tsx:300]: You're missing `fetchError` in the condition

File: packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx [hunk 7]
```diff
@@ -148,6 +148,8 @@ export const ArrayInput = (props: ArrayInputProps) => {
             </Labeled>
         );
     }
+    const renderHelperText =
+        helperText !== false || ((isDirty || isSubmitted) && !!error);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx:152]: This condition is different from the one before.

If `helperText = undefined` (the default):
- Before we would have rendered `null`
- Now we render `<FormHelperText ... />`

This creates an unnecessary additional gap.
You should keep the previous condition IMO.

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx (hunk 7 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx:152]: It was wrong before. The additionnal gap isn't unnecessary as it prevents the UI from jumping when error messages are displayed

**Review by slax57 — APPROVED** [review]

**Review by djhi — APPROVED** [review]

**Pull request merged**